ipykernel/kernelbase.py

```diff
@@ -975,7 +975,7 @@ def raw_input(self, prompt=''):
         return self._input_request(
             str(prompt),
             self._parent_ident["shell"],
-            self.get_parent_header("shell"),
+            self.get_parent("shell"),
```

Comment:
The value returned by `get_parent("shell")` is the full parent message or just the header?

Reply:
Ah, the answer to my question was in the changes in the alternative pr. Merging!